protocols/bgp/server/server.go

```diff
@@ -51,6 +51,10 @@ func (b *bgpServer) RouterID() uint32 {
 }
 
 func (b *bgpServer) Start(c *config.Global) error {
+	if err := c.ReadGlobalConfig(); err != nil {
```

Comment:
The BGP server must be independent of the config file in order to be usable as a library.
Config MUST be read outside of protocol (be it BGP, BMP or ISIS) structures and be passed as an argument.